controllers/servingruntime_controller.go

```diff
@@ -369,23 +414,47 @@ func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Co
 		return nil, err
 	}
 
+	// list all cluster serving runtimes
+	csrs := &kserveapi.ClusterServingRuntimeList{}
+	if r.HasNamespaceAccess {
+		if err := r.Client.List(ctx, csrs); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
nit: could move this below and include the `csrs` declaration and for loop within the if